modules/openid_auth/app/controllers/openid_auth/v2/validation_controller.rb

```diff
@@ -34,6 +35,8 @@ def validated_payload
           payload_object.launch[:patient] = @current_user.icn
         end
 
+        payload_object.act[:type] = 'system' if token.client_credentials_token?
+
```

Comment:
One thought to simplify things is move the `if token.client_credentials_token?` check up to line 31 and return the payload similar to the ssoi_token? block.

Then that `unless` block could be reduced to a simple `if payload_object[:icn].nil?`.

Reply:
I think the issue here is that it is possible for a non client_credentials token to have null ICN. In that case we would be setting the wrong act.type. Or we would put that check about the null icn check and we would have dubplicated code between the client_credentials token statement and the null icn statement.